src/controllers/TalksController.php

```diff
@@ -498,12 +498,88 @@ protected function getTalkDataFromRequest(PDO $db, Request $request, $event_id)
             FILTER_SANITIZE_URL
         );
 
-        $talk['speakers'] = array_map(function ($speaker) {
-            $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
-            $speaker = trim($speaker);
-            return $speaker;
-        }, (array) $request->getParameter('speakers'));
+        $talk['speakers'] = array_map(
+            function ($speaker) {
+                $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
+                $speaker = trim($speaker);
+                return $speaker;
+            },
+            (array) $request->getParameter('speakers')
+        );
+
+        return $talk;
+    }
+
+    public function getSpeakersForTalk(Request $request, PDO $db)
+    {
+        $talk_id = $this->getItemId($request);
+        $talk = $this->getTalkById($db, $request, $talk_id);
+        return $talk->speakers;
+    }
+
+    public function setSpeakerForTalk(Request $request, PDO $db)
+    {
+        //We must be logged in
+        if (!isset($request->user_id)) {
+            throw new Exception("You must be logged in to create data", 401);
+        }
+
+        $talk_id = $this->getItemId($request);
+        $talk_mapper = new TalkMapper($db, $request);
+        $talk = $talk_mapper->getTalkById($talk_id);
+        if (! $talk) {
+            throw new Exception("Talk not found", 404);
```

Comment:
Not sure what the policy is for Joind.in regarding this, but I'd encourage using specific exception classes - at minimum, the SPL exceptions could be used instead. Apologies if this is an already-discussed point, or if there is an import that I missed